kernel/uid_gid.c

```diff
@@ -0,0 +1,775 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#define _GNU_SOURCE
+#include <limits.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <myst/kernel.h>
+#include <myst/syscall.h>
+#include <myst/thread.h>
+
+/* Currently only supports a single mapping which is typically from enclave root
+ * identity to the host application identity when executed. */
+static myst_host_enc_id_mapping host_enc_id_mapping;
+
+void myst_set_host_uid_gid_mapping(myst_host_enc_id_mapping id_mapping)
+{
+    host_enc_id_mapping = id_mapping;
+}
+
+uid_t myst_enc_uid_to_host(uid_t euid)
+{
+    if (euid == host_enc_id_mapping.enc_uid)
+        return host_enc_id_mapping.host_uid;
+    else
+        return euid;
```

Comment:
It doesn't sound right to return enclave UID as the host UID if no mapping is found. Should we error out or return host_uid anyway?

Reply:
no. At some point we will an array of mappings. currently the tests use root which needs a specific mapping, but other standard users currently are the same inside and outside the enclave (nobody, bin, daemon, etc...)